wfe/wfe.go

```diff
@@ -1365,7 +1354,7 @@ func (wfe *WebFrontEndImpl) Order(
 	request *http.Request) {
 
 	var account *core.Account
-	if request.Method == "GET" && wfe.strict {
+	if request.Method == "GET" {
```

Comment:
I don't think this check is meaningful anymore because the handle func is only registered for "POST" now:
> wfe.HandleFunc(m, orderPath, wfe.Order, "POST")

Reply:
I think this also means the `else if` that follows can be simplified.